File: tools/test_golden.ts
import * as fs from 'fs';
import * as path from 'path';
import { FluxConstraint, Severity } from '../src/typescript/flux-constraint';

const vectors = JSON.parse(fs.readFileSync(path.join(__dirname, 'golden_vectors.json'), 'utf-8'));
let mismatches = 0;

for (const v of vectors) {
    const cs = v.constraints.map((c: any) => ({ lo: c.lo, hi: c.hi, name: '' }));
    const fc = new FluxConstraint(cs);
    const r = fc.check(v.value);
    const exp = v.expected;
    if (r.errorMask !== exp.error_mask || r.passed !== exp.passed || r.violatedCount !== exp.violated_count) {
        mismatches++;
        if (mismatches <= 5) console.log(`MISMATCH #${v.id}: value=${v.value} got mask=${r.errorMask} passed=${r.passed} expected mask=${exp.error_mask}`);
    }
}

console.log(`\nTypeScript: ${vectors.length} vectors, ${mismatches} mismatches`);
process.exit(mismatches > 0 ? 1 : 0);

File: tools/test_golden.dart
import 'dart:io';
import 'dart:convert';
import '../src/dart/flux_constraint.dart';

void main() {
  final vectors = jsonDecode(File('tools/golden_vectors.json').readAsStringSync()) as List;
  int mismatches = 0;

  for (final v in vectors) {
    final cs = (v['constraints'] as List).map((c) => Constraint(c['lo'], c['hi'], '')).toList();
    final fc = FluxChecker(cs);
    final r = fc.check(v['value']);
    final exp = v['expected'];
    if (r.errorMask != exp['error_mask'] || r.passed != exp['passed'] || r.violatedCount != exp['violated_count']) {
      mismatches++;
      if (mismatches <= 5) print('MISMATCH #${v["id"]}: value=${v["value"]} got mask=${r.errorMask}');
    }
  }

  print('\nDart: ${vectors.length} vectors, $mismatches mismatches');
  exit(mismatches > 0 ? 1 : 0);
}
